cfme/infrastructure/provider/__init__.py

```diff
@@ -167,12 +167,14 @@ def hosts(self):
         result = []
         host_collection = self.appliance.collections.hosts
         for host in self.data.get("hosts", []):
-            creds = conf.credentials.get(host["credentials"], {})
-            cred = Host.Credential(
-                principal=creds["username"],
-                secret=creds["password"],
-                verify_secret=creds["password"],
-            )
+            cred = {}
+            for cred_type in host["credentials"]:
```

Comment:
**required** the `items()` trick
**optional** this one has no conditionals and might qualify for a dict comprehension expression